tutorials/03_plugins.md

```diff
@@ -3,121 +3,216 @@
+{
+}
+```
+
+The destructor is empty because no cleanup is required as such.
+
+```cpp
+HelloPlugin::~HelloPlugin()
+{
+}
+```
+
+### The Configuration File
 
-### Topic viewer
+A plugin may require certain configuration files to specify certain parameter values.
 
-**Not ported to versions 1 or higher yet**
+In this case, we would want the configuration file to hold the message to be printed on the display (here the display is the terminal) when the `Hello, Plugin!` button is clicked.
 
-Discover all Gazebo Transport topics and introspect its fields.
+Thus, the [HelloPlugin.config](https://github.com/gazebosim/gz-gui/blob/main/examples/plugin/hello_plugin/HelloPlugin.config) looks like this:
 
-    gz gui -c examples/config/topicViewer.config
+```xml
+<plugin filename="HelloPlugin">
+  <message>Hello, plugin config!</message>
+</plugin>
+```
 
-### Topics stats
+The `LoadConfig` function tries to get the message from the message tag.
 
-**Not ported to versions 1 or higher yet**
+```cpp
+void HelloPlugin::LoadConfig(const tinyxml2::XMLElement *_pluginElem)
+{
+  if (!_pluginElem)
+    return;
 
-Show basic stats of all Gazebo Transport topics at the same time.
+  // Take parameters from XML at runtime
+  auto messageElem = _pluginElem->FirstChildElement("message");
+  if (nullptr != messageElem && nullptr != messageElem->GetText())
+    this->message = messageElem->GetText();
+}
+```
 
-    gz gui -c examples/config/topicsStats.config
+We have the `OnButton()` function which would print the message when clicked.
 
-### Requester and Responder
+```cpp
+void HelloPlugin::OnButton()
+{
+  std::cout << this->message << std::endl;
+}
+```
 
-**Not ported to versions 1 or higher yet**
+Thereafter, `GZ_ADD_PLUGIN` registers the plugin with the Gazebo GUI system, making it discoverable at runtime.
 
-Make custom service requests and serve custom services with these two
-plugins.
+Refer to the [documentation](https://gazebosim.org/api/plugin/2/gz_2plugin_2Register_8hh.html).
 
-    gz gui -c examples/config/reqres.config
+```cpp
+GZ_ADD_PLUGIN(HelloPlugin,
+              gz::gui::Plugin);
+```
 
-### Displays
+## Configuring plugins
 
-**Not ported to versions 1 or higher yet**
+As shown above, Gazebo GUI supports loading XML configuration files, which may contain
+parameters to be passed directly to the plugin. See the \ref config
+tutorial for more information.
 
-This is a container for various Display Plugins that each paint on a Scene.
-The following example includes the display plugins that are installed with
-Gazebo GUI by default.
+## Finding plugins
 
-    gz gui -c examples/config/display_plugins.config
+Gazebo GUI will look for plugins on the following paths, in this order:
+
+1. All paths set on the `GZ_GUI_PLUGIN_PATH` environment variable
+2. All paths added by calling `gz::gui::addPluginPath`
+3. `~/.gz/gui/plugins`
+4. [Plugins which are installed with Gazebo GUI](https://gazebosim.org/api/gui/9/namespacegz_1_1gui_1_1plugins.html)
```

Comment:
the link actually does not list any plugins. You can replace it with `https://github.com/gazebosim/gz-gui/tree/gz-gui9/src/plugins`